protocol/x/vault/keeper/withdraw.go

```diff
@@ -42,3 +44,100 @@ func (k Keeper) ValidateWithdrawFromVault(
 
 	return nil
 }
+
+func (k Keeper) RedeemShares(
+	ctx sdk.Context,
+	vaultId types.VaultId,
+	owner string,
+	sharesToRedeem types.NumShares,
+) (quantumsToWithdraw *big.Int, err error) {
+	// **** Validate ****
+	// Validate that `sharesToRedeem` is positive.
+	sharesToRedeemInt := sharesToRedeem.NumShares.BigInt()
+	if sharesToRedeemInt.Sign() <= 0 {
+		return nil, errors.Wrapf(
+			types.ErrInvalidWithdrawalAmount,
+			"Shares to withdraw must be positive",
+		)
+	}
+	totalShares, vaultExists := k.GetTotalShares(ctx, vaultId)
+	totalShareInt := totalShares.NumShares.BigInt()
+	ownerShares, ownerSharesExist := k.GetOwnerShares(ctx, vaultId, owner)
+	ownerSharesInt := ownerShares.NumShares.BigInt()
+	// Validate that vault and owner shares exist.
+	if !vaultExists || !ownerSharesExist {
+		return nil, errors.Wrapf(
```

Comment:
nit: should we include `totalShares` and `ownerShares` in this error as well to tell whether it's the vault that doesn't exist or the owner that doesn't exist? if non-existent, it'll just be `nil`